### PR TITLE
fix: allow for updates of supplier name and url

### PR DESCRIPTION
**Pull request by CBeck-96**
Allows for updates of supplier.url AND supplier.name.
Also contains the change to the documentation.

### CODE REVIEW

File: docs/available_commands.md [hunk 1]
```diff
@@ -14,10 +14,10 @@ Currently, the command adds or modifies the following pieces of information:
 
 * If the SBOM metadata doesn't specify an *author* from the SBOM, it will be set to `{"name": "automated"}`.
 * The *compositions* array will be overwritten with a new one which specifies a single *incomplete* aggregate. This aggregate contains all components, including the metadata component.
-* If a component does not have a *supplier*, the tool will try to infer the supplier from the fields (in order of precedence):
+* If a component does have a publisher and/or author but does not have a *supplier*, the tool will try to infer the `supplier.name` from the fields (in order of precedence):
```
Comment (italvi) [inline docs/available_commands.md:17]: for consistency, I would either change the "a component" to "the component" or revert the change to "the" you did on [L#20](https://github.com/Festo-se/cyclonedx-editor-validator/pull/118/files#diff-bea07d8bee938360185b0b313b81496c7d1c302766e3d55def5f34279c161023R20)

File: cdxev/amend/operations.py [hunk 2]
```diff
@@ -119,53 +119,62 @@ class InferSupplier(Operation):
     """
     As we need a contact in case of a security incident, at least one of the 'author', 'supplier'
     or 'publisher' fields must be set on any component. If that's not the case, this operation
```
Comment (italvi) [inline cdxev/amend/operations.py:121]: As you are now just try to add further information, I would change the reasoning

File: cdxev/amend/operations.py [hunk 3]
```diff
@@ -119,53 +119,62 @@ class InferSupplier(Operation):
...
+                            "Set supplier of %s to URL: %s",
+                            component.get("bom-ref", "<no bom-ref>"),
+                            ext_ref["url"],
+                        )
+                        break
+
+        if "name" not in component.get("supplier", {}):
+
+            if "publisher" in component:
+                component["supplier"] = component.get("supplier", {})
```
Comment (italvi) [inline cdxev/amend/operations.py:170]: maybe add this line before the two `if`s?

File: tests/test_amend.py [hunk 4]
```diff
@@ -179,6 +179,68 @@ def test_author_set_supplier_components(self) -> None:
             self.sbom_fixture["components"][0]["author"],
         )
 
+    def test_supplier_gets_not_overwritten(self) -> None:
```
Comment (italvi) [inline tests/test_amend.py:182]: what is the difference between this test and `test_supplier_name_url_get_no_overwritten`?

**Review by italvi — CHANGES_REQUESTED** [review]

File: cdxev/amend/operations.py (hunk 3 above)
Comment (CBeck-96) [reply cdxev/amend/operations.py:170]: Then this might result in an empty 'supplier' getting created.

File: cdxev/amend/operations.py (hunk 2 above)
Comment (CBeck-96) [reply cdxev/amend/operations.py]: Rewrote it
